File: packages/typescript/src/resources/compliance.ts
/**
 * The `compliance` resource: per-country sending rules and content rules.
 * @module
 */
import type { ContentRule, CountryRules, RequestOptions } from '../models.js';
import { Resource, seg } from './base.js';

/** Accessed as `client.compliance`. */
export class Compliance extends Resource {
  listCountries(options?: RequestOptions): Promise<CountryRules[]> {
    return this.array({ method: 'GET', path: '/v1/compliance/countries' }, options);
  }

  getCountry(iso2: string, options?: RequestOptions): Promise<CountryRules> {
    return this.one({ method: 'GET', path: `/v1/compliance/countries/${seg(iso2, 'iso2')}` }, options);
  }

  listContentRules(options?: RequestOptions): Promise<ContentRule[]> {
    return this.array({ method: 'GET', path: '/v1/content-rules' }, options);
  }
}

File: packages/rust/src/resources/compliance.rs
//! The `compliance` resource: country rules and content rules.

use super::seg;
use crate::error::Result;
use crate::http::HttpClient;
use crate::models::{ContentRule, CountryRules};

/// Accessed as `client.compliance()`.
#[derive(Debug, Clone)]
pub struct Compliance {
    http: HttpClient,
}

impl Compliance {
    pub(crate) fn new(http: HttpClient) -> Self {
        Self { http }
    }

    /// Rules for every country.
    pub async fn list_countries(&self) -> Result<Vec<CountryRules>> {
        self.http.get("/v1/compliance/countries").await
    }

    /// Rules for one country (ISO2).
    pub async fn get_country(&self, iso2: &str) -> Result<CountryRules> {
        self.http
            .get(&format!("/v1/compliance/countries/{}", seg(iso2, "iso2")?))
            .await
    }

    /// Content rules that apply to the workspace.
    pub async fn list_content_rules(&self) -> Result<Vec<ContentRule>> {
        self.http.get("/v1/content-rules").await
    }
}
